spec/extensions/fourier_transform_functions.md

```diff
@@ -0,0 +1,505 @@
+# Fourier transform Functions
+
+> Array API specification for Fourier transform functions.
+
+A conforming implementation of the array API standard must provide and support the following functions adhering to the following conventions.
+
+-   Positional parameters must be [positional-only](https://www.python.org/dev/peps/pep-0570/) parameters. Positional-only parameters have no externally-usable name. When a function accepting positional-only parameters is called, positional arguments are mapped to these parameters based solely on their order.
+-   Optional parameters must be [keyword-only](https://www.python.org/dev/peps/pep-3102/) arguments.
+-   Broadcasting semantics must follow the semantics defined in {ref}`broadcasting`.
+-   Unless stated otherwise, functions must support the data types defined in {ref}`data-types`.
+-   Unless stated otherwise, functions must adhere to the type promotion rules defined in {ref}`type-promotion`.
+-   Unless stated otherwise, floating-point operations must adhere to IEEE 754-2019.
+
+## Objects in API
+
+<!-- NOTE: please keep the functions and their inverse together -->
+(function-fft)=
+### fft(a, /, *, n=None, axis=-1, norm='backward')
+
+Computes the one-dimensional discrete Fourier transform. The expected behavior includes a round-trip transform using the inverse function, `fft(ifft(a)) == a`.
+
+#### Parameters
+
+-   **a**: _&lt;array&gt;_
+
+    -   Input array.
```

Comment:
We should indicate the required dtypes for inputs and the expected dtypes for the outputs (see other parts of the spec, for instance, elementwise functions, for how this is generally worded).

Reply:
Aha! @asmeurer good catch. I missed that we need to specify the type casting rules. `numpy.fft` did a bad decision to always return in double precision, which we (CuPy) did not appreciate and deviated on purpose:
https://docs.cupy.dev/en/stable/reference/fft.html#code-compatibility-features
Seriously, with all these shiny backends (pocketfft, FFTW, cuFFT, ...) this is really not necessary.

Reply:
Should we always expect single precision for the output? Or should it depend on the input dtype?

Reply:
Sorry I dropped the ball, Stephannie. I'd say for floating-number (real or complex) inputs we keep the same precision, and reject integer inputs as done elsewhere in the standard.